src/librustc_parse/parser/diagnostics.rs

```diff
@@ -934,6 +934,25 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
```

Comment:
What happens if we instead change this to the following and remove the new code?
```suggestion
        } else if !sm.is_multiline(self.prev_token.span.between(self.token.span)) {
```

Reply:
No change from OP with this change:

```
error: expected `;`, found `,`
 --> src/main.rs:4:5
  |
4 |     ,arg("2")
  |     ^ help: change this to `;`: `;`

error[E0425]: cannot find function `arg` in this scope
 --> src/main.rs:4:6
  |
4 |     ,arg("2")
  |      ^^^ not found in this scope

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0425`.
error: could not compile `tester`.

To learn more, run the command again with --verbose.
```